src/main/java/org/truffleruby/core/rope/RepeatingRope.java

```diff
@@ -33,7 +33,7 @@ public Rope withEncoding(Encoding newEncoding, CodeRange newCodeRange) {
 
     @Override
     public boolean canFastEncode(Encoding newEncoding, CodeRange newCodeRange) {
-        return true;
+        return child.canFastEncode(newEncoding, newCodeRange);
```

Comment:
I don't think this is wrong, but it should always end up being `return newCodeRange == getCodeRange();`, just like every other managed rope. Conceptually, this case is no different than `SubstringRope`.

If the issue is the child is a native rope, then we shouldn't allow that. We can't compose native ropes and must convert to managed ropes when composing them.

Reply:
Looking again briefly, `RopeBuffer` is considered a managed rope. I've decided I'm just going to rip that code out and replace it with something that fits the new hierarchy better. Conceptually, it's closer to a native rope than a leaf rope.

Reply:
#564 was merged, which removes RopeBuffer. You'll have to rebase I'm afraid, but it should simplify a lot of the code here I think.